pymc/distributions/timeseries.py

```diff
@@ -293,37 +197,86 @@ def dist(cls, mu=0.0, sigma=1.0, *, init_dist=None, steps=None, **kwargs) -> at.
 
         return super().dist([mu, sigma, init_dist, steps], **kwargs)
 
-    def moment(rv, size, mu, sigma, init_dist, steps):
-        grw_moment = at.zeros_like(rv)
-        grw_moment = at.set_subtensor(grw_moment[..., 0], moment(init_dist))
-        # Add one dimension to the right, so that mu broadcasts safely along the steps
-        # dimension
-        grw_moment = at.set_subtensor(grw_moment[..., 1:], mu[..., None])
-        return at.cumsum(grw_moment, axis=-1)
-
-    def logp(
-        value: at.Variable,
-        mu: at.Variable,
-        sigma: at.Variable,
-        init_dist: at.Variable,
-        steps: at.Variable,
-    ) -> at.TensorVariable:
-        """Calculate log-probability of Gaussian Random Walk distribution at specified value."""
-
-        # Calculate initialization logp
-        init_logp = logp(init_dist, value[..., 0])
-
-        # Make time series stationary around the mean value
-        stationary_series = value[..., 1:] - value[..., :-1]
-        # Add one dimension to the right, so that mu and sigma broadcast safely along
-        # the steps dimension
-        series_logp = logp(Normal.dist(mu[..., None], sigma[..., None]), stationary_series)
-
-        return check_parameters(
-            init_logp + series_logp.sum(axis=-1),
-            steps > 0,
-            msg="steps > 0",
+    @classmethod
+    def ndim_supp(cls, *args):
+        return 1
+
+    @classmethod
+    def rv_op(cls, mu, sigma, init_dist, steps, size=None):
+        mu_shape = get_folded_shape(mu.shape)
+        sigma_shape = get_folded_shape(sigma.shape)
+        init_dist_shape = get_folded_shape(init_dist.shape)
+
+        # If not explicit, size is determined by the shapes of mu, sigma, and init
+        if size is not None:
+            # we have all the information regarding size from users or .dist()
+            init_size = size
+        else:
+            # we infer size from parameters
+            init_size = at.broadcast_shape(
+                mu_shape,
+                sigma_shape,
+                init_dist_shape,
+                arrays_are_shapes=True,
+            )
+
```

Comment:
I'm not so sure if `ValueError: Random variables detected in the logp graph` arises due to inference of distribution shapes. Here is my train of thought:

- I was able to retrieve shapes via `FunctionGraph`, but the same error still occurs.
- If we specify the `size` argument, the same error occurs as well.
- In my `test_moment`, I substituted the `init_dist` by a Student T distribution and normal rvs were still present in the logp graph

Would the issue be inherent to the use of AePPL and how `joint_logp` is written in `logprob.py`? It's been a while, I may have forgotten some important details...

Reply:
Hmm... interesting. We need to figure out when are these RVs included in the graph then, and if they correspond to other model variables why aren't they being replaced by their value variables in aeppl.

Can you share a small snippet that raises the error?

It might also help to create the GRW manually (just with aesara.tensor calls and not pymc) and give names to the intermediate variables (cumsum, join, etc ...). You can include your manual GRW RV in a model by calling `model.register_rv`

Reply:
Here is a notebook reproducing the issue: https://github.com/larryshamalama/aesara-playground/blob/main/notebooks/grw-rv-logp-graph.ipynb

I believe that a normal rv is present with id CT